backend/census_historical_migration/workbooklib/federal_awards.py

```diff
@@ -252,6 +252,76 @@ def xform_missing_cluster_total(
         InspectionRecord.append_federal_awards_changes(change_records)
 
 
+def xform_missing_cluster_total_v2(
+    audits,
+    cluster_names,
+    other_cluster_names,
+    state_cluster_names,
+):
+    """Calculates missing cluster total for each award."""
+    uniform_state_cluster_names = [name.upper() for name in state_cluster_names]
+    uniform_other_cluster_names = [name.upper() for name in other_cluster_names]
+    amounts_expended = [
+        string_to_int(audit.AMOUNT) if string_to_string(audit.AMOUNT) else 0
+        for audit in audits
+    ]
+
+    change_records = []
+    invalid_records = []
+    is_empty_cluster_total_found = False
+    is_incorrect_cluster_total_found = False
+
+    for idx, (name, audit) in enumerate(zip(cluster_names, audits)):
+        cluster_total = string_to_string(audit.CLUSTERTOTAL)
+        if not cluster_total:
+            is_empty_cluster_total_found = True
+        else:
+            cluster_total = string_to_int(cluster_total)
+
+        gsa_cluster_total = expected_cluster_total(
+            idx=idx,
+            name=name,
+            uniform_other_cluster_names=uniform_other_cluster_names,
+            uniform_state_cluster_names=uniform_state_cluster_names,
+            state_cluster_names=state_cluster_names,
+            cluster_names=cluster_names,
+            amounts_expended=amounts_expended,
+        )
+
+        if not is_empty_cluster_total_found and cluster_total != gsa_cluster_total:
+            is_incorrect_cluster_total_found = True
+        else:
+            cluster_total = gsa_cluster_total
+
+        track_invalid_records(
+            [("CLUSTERTOTAL", cluster_total)],
```

Comment:
We need to capture the other fields used in the calculation of cluster_total

Reply:
Done